codebuild/bin/s2n_codebuild.sh

```diff
@@ -41,13 +41,23 @@ fi
 make clean;
 
 echo "Using $JOBS jobs for make..";
+echo "running with libcrypto: ${S2N_LIBCRYPTO}, gcc_version: ${GCC_VERSION}"
 
 if [[ "$OS_NAME" == "linux" && "$TESTS" == "valgrind" ]]; then
     # For linux make a build with debug symbols and run valgrind
     # We have to output something every 9 minutes, as some test may run longer than 10 minutes
     # and will not produce any output
     while sleep 9m; do echo "=====[ $SECONDS seconds still running ]====="; done &
-    S2N_DEBUG=true make -j $JOBS valgrind
+
+    if [[ "$S2N_LIBCRYPTO" == "openssl-1.1.1" || "$S2N_LIBCRYPTO" == "awslc" ]]; then
```

Comment:
as this [CI run](https://us-west-2.console.aws.amazon.com/codesuite/codebuild/024603541914/projects/s2nGeneralBatch/batch/s2nGeneralBatch%3Aa3b8ea14-ab8d-4906-81c2-8221ec09f81d?region=us-west-2) shows, `awslc` is also passing.

 I have identified the failing Valgrind runs and documented a bit about the effort involved in fixing the remaining runs in the tracking issue: https://github.com/aws/s2n-tls/issues/3758